conan_provider.cmake

```diff
@@ -50,7 +50,7 @@ function(detect_os OS OS_API_LEVEL OS_SDK OS_SUBSYSTEM OS_VERSION)
                 message(STATUS "CMake-Conan: cmake_osx_sysroot=${CMAKE_OSX_SYSROOT}")
                 set(${OS_SDK} ${_OS_SDK} PARENT_SCOPE)
             endif()
-            if(DEFINED CMAKE_OSX_DEPLOYMENT_TARGET)
+            if(DEFINED CONAN_CUSTOM_CMAKE_OSX_DEPLOYMENT_TARGET)
```

Comment:
It is a bit weird that the variable is given the value of the original content, but then it is not used. If it is intended to be used as a boolean toggle, then maybe when it is assigned below, assign it directly to True/On?

Reply:
Agreed! I would either:
* Make it a bool (along the lines of `CONAN_CMAKE_OSX_DEPLOYMENT_AUTODETECTED`)
* or save the value, in case _somehow_ the user does a `set(CMAKE_OSX_DEPLOYMENT_TARGET ...)` _after_ project (i know, too many cases) - then we can compare. Although we don't have to cover this case, if the documentation says it should be cache variable, then it should!